src/Microsoft.NetCore.Analyzers/Core/Security/SetHttpOnlyForHttpCookie.cs

```diff
@@ -0,0 +1,205 @@
+                                        propertyReferenceOperation.Property.Name == "HttpOnly" &&
+                                        simpleAssignmentOperation.Value.ConstantValue.HasValue &&
+                                        simpleAssignmentOperation.Value.ConstantValue.Value.Equals(false))
+                                    {
+                                        operationAnalysisContext.ReportDiagnostic(
+                                            simpleAssignmentOperation.CreateDiagnostic(
+                                                Rule));
+                                    }
+                                },
+                                OperationKind.SimpleAssignment);
+
+                            operationBlockStartAnalysisContext.RegisterOperationAction(
+                                (OperationAnalysisContext operationAnalysisContext) =>
+                                {
+                                    IObjectCreationOperation objectCreationOperation = (IObjectCreationOperation)operationAnalysisContext.Operation;
+                                    if (httpCookieSymbol.Equals(objectCreationOperation.Type))
+                                    {
+                                        ImmutableArray<IArgumentOperation> arguments = objectCreationOperation.Arguments;
+                                        if (arguments.Length < 2 ||
+                                            !arguments[0].Value.ConstantValue.HasValue ||
+                                            arguments[0].Value.ConstantValue.Value == null ||
+                                            !arguments[0].Value.ConstantValue.Value.Equals("x-ms-cpim-trans") ||
+                                            !arguments[1].Value.ConstantValue.HasValue ||
+                                            arguments[1].Value.ConstantValue.Value == null ||
+                                            !arguments[1].Value.ConstantValue.Value.Equals("Invalid_Base64_String"))
+                                        {
+                                            return;
+                                        }
+
+                                        operationAnalysisContext.ReportDiagnostic(
+                                            objectCreationOperation.CreateDiagnostic(
+                                                Rule));
+                                    }
+                                },
+                                OperationKind.ObjectCreation);
+
+                            operationBlockStartAnalysisContext.RegisterOperationAction(
+                                (OperationAnalysisContext operationAnalysisContext) =>
+                                {
+                                    IReturnOperation returnOperation = (IReturnOperation)operationAnalysisContext.Operation;
+                                    if (httpCookieSymbol.Equals(returnOperation.ReturnedValue.Type))
```

Comment:
`IReturnOperation.ReturnedValue` can be `null` for return statement without value for error cases.